backend/private-graph/graph/middleware.go

```diff
@@ -109,6 +119,62 @@ func (c *SimpleAuthClient) GetUser(_ context.Context, _ string) (*auth.UserRecor
 	}, nil
 }
 
+func authenticateToken(tokenString string) error {
+	claims := jwt.MapClaims{}
+	_, err := jwt.ParseWithClaims(tokenString, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(JwtAccessSecret), nil
+	})
+	if err != nil {
+		return e.Wrap(err, "invalid id token")
+	}
+
+	exp, ok := claims["exp"].(float64)
+	if !ok {
+		return e.Wrap(err, "invalid exp claim")
+	}
+
+	// Check if the current time is after the expiration
+	if time.Now().After(time.Unix(int64(exp), 0)) {
+		return e.Wrap(err, "token expired")
+	}
+
+	return nil
+}
+
+func (c *PasswordAuthClient) GetUser(_ context.Context, _ string) (*auth.UserRecord, error) {
+	return &auth.UserRecord{
+		UserInfo: &auth.UserInfo{
+			DisplayName: "Hobby Highlighter",
+			Email:       "demo@example.com",
+			PhoneNumber: "+14081234567",
+			PhotoURL:    "https://picsum.photos/200",
+			ProviderID:  "",
+			UID:         "12345abcdef09876a1b2c3d4e5f",
+		},
+		EmailVerified: true,
+	}, nil
+}
+
+func (c *PasswordAuthClient) updateContextWithAuthenticatedUser(ctx context.Context, token string) (context.Context, error) {
```

Comment:
Same as implementation for SimpleAuthClient but token is validated

Reply:
nice!